conans/model/build_info.py

```diff
@@ -490,22 +489,24 @@ def get_sorted_components(self):
 
     def aggregated_components(self):
         """Aggregates all the components as global values, returning a new CppInfo"""
-        if self._aggregated is None:
```

Comment:
Asking myself if there are more instances of caching in the code that can also be an issue

Reply:
I had a quick look, there is caching in some other places related to the computation of the graph, but nothing evident at first sight that would affect the generators